### PR TITLE
Implement the module to create `AWS EC2 Container Registry`

### PR DESCRIPTION
**Pull request by s2504s**
## What
* Implement the TF module to create `AWS EC2 Container Registry`

## Why
* We need this in order to store our images for Docker containers

## Reference
* Based on https://bitbucket.org/cultivatedops/terraform-module-aws-ecr
* Docker usage https://aws.amazon.com/blogs/compute/authenticating-amazon-ecr-repositories-for-docker-cli-with-credential-helper/

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  namespace = "${var.namespace}"
+  stage     = "${var.stage}"
+  name      = "${var.name}"
+}
+
+resource "aws_ecr_repository" "default" {
+  name = "${module.label.id}"
+}
+
+resource "aws_ecr_repository_policy" "policy" {
```
Comment (osterman) [inline main.tf]: rename to `default`

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,75 @@
+data "aws_iam_policy_document" "ecr" {
```
Comment (osterman) [inline main.tf]: rename to `default`

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  statement {
+    sid    = "user"
+    effect = "Allow"
+
+    principals {
+      type = "AWS"
+
+      identifiers = [
```
Comment (osterman) [inline main.tf]: Can we add support for groups ARNs?

**Comment by osterman** [comment]
- [ ] Fix `LICENSE`

File: output.tf [hunk 4]
```diff
@@ -0,0 +1,24 @@
+output "arn" {
+  value = "${aws_ecr_repository.default.arn}"
+}
+
+//
+//output "name" {
```
Comment (osterman) [inline output.tf]: I think you should uncomment this

Comment (s2504s) [inline output.tf]: Done

File: variables.tf [hunk 5]
```diff
@@ -0,0 +1,18 @@
+variable "name" {
+  description = "Name of the repository"
+  default     = ""
```
Comment (osterman) [inline variables.tf]: Let's remove the default values for `name`, `namespace`, and `stage`

Comment (s2504s) [reply variables.tf]: Removed

**Comment by osterman** [comment]
Ok, my confusion was the unnecessary logic around
```
${signum(length(var.roles)) == 1 ? 1 : 0}
```

which is identical to 
```
${signum(length(var.roles))}
```

I don't think there's any need to add a ternary unless inverting the outcome.

**Pull request merged**